src/python/pants/util/collections.py

```diff
@@ -2,23 +2,21 @@
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
 import collections
+from typing import Callable, DefaultDict, Iterable, MutableMapping, Tuple, TypeVar
 
 
-def combined_dict(*dicts):
-  """Combine one or more dicts into a new, unified dict (dicts to the right take precedence)."""
-  return {k: v for d in dicts for k, v in d.items()}
```

Comment:
There's no use for this function anymore thanks to dictionary unpacking: `{**d1, **d2}`